From 76e407faf90641e644479463c7231fba46fcb439 Mon Sep 17 00:00:00 2001
From: Ilias Garnier <igarnier@protonmail.com>
Date: Tue, 3 Oct 2023 20:25:19 +0200
Subject: [PATCH] Produce static archive

---
 Makefile | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Makefile b/Makefile
index b8b0d4b..71b8e6f 100644
--- a/Makefile
+++ b/Makefile
@@ -532,6 +532,9 @@ train.o: common/train.cpp common/train.h
 libllama.so: llama.o ggml.o $(OBJS)
 	$(CXX) $(CXXFLAGS) -shared -fPIC -o $@ $^ $(LDFLAGS)

+libllama.a: llama.o ggml.o $(OBJS)
+	ar -rc $@ $^
+
 clean:
 	rm -vrf *.o tests/*.o *.so *.dll benchmark-matmult build-info.h *.dot $(COV_TARGETS) $(BUILD_TARGETS) $(TEST_TARGETS)

--
2.34.1
