From 75785d11eaa929f2196fa9862e244ec4a817fdea Mon Sep 17 00:00:00 2001
From: Ilias Garnier <igarnier@protonmail.com>
Date: Sat, 30 Sep 2023 07:07:55 +0200
Subject: [PATCH] patching Makefile

---
 Makefile | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Makefile b/Makefile
index 08b83ca..59b90f1 100644
--- a/Makefile
+++ b/Makefile
@@ -530,6 +530,9 @@ train.o: common/train.cpp common/train.h
 libllama.so: llama.o ggml.o $(OBJS)
 	$(CXX) $(CXXFLAGS) -shared -fPIC -o $@ $^ $(LDFLAGS)

+libllama.a: llama.o ggml.o $(OBJS)
+	ar -rc $@ $^
+
 clean:
 	rm -vrf *.o tests/*.o *.so *.dll benchmark-matmult build-info.h *.dot $(COV_TARGETS) $(BUILD_TARGETS) $(TEST_TARGETS)

--
2.34.1
